### PR TITLE
Add README

### PR DESCRIPTION
**Pull request by ezmiller**
@thomastuts and @AnnJohn this is ready for review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,177 @@
...
+...bad? Actually, no. It's not that state is bad. In fact, using state in our React components is often
+necessary, and the ability to explicitly and reliably update state in our components
+is one of the most powerful features of React. But, state creates complexity.
 
-Props are purely declarative. So if we can move state to the edges of our
-application — say, only updating a parent view and passing the state down as
-props to child components — we can greatly reduce that complexity (and make
-our components dead-simple to test).
+Once a component has state, it becomes harder to predict its output and behavior.
+Because, by definition, state changes, we won't be able to say at any given point
```
Comment (thomastuts) [inline README.md]: 'Because state changes by definition, ...'?

File: README.md [hunk 2]
```diff
@@ -1,30 +1,177 @@
...
+in our program's operation what state our component is in. That, in turn, will
+make it harder to reason about our application; and when it becomes harder to
+reason about our application we will struggle more to eradicate and avoid bugs.
 
-This is one of the main ideas behind `connect()` (previously, `@connect()`)
-in Redux: we can abstract away state management so that we're only dealing
-with props. We still don't quite want to introduce students to flux or Redux
-quite yet, but we want to get them a bit closer.
+> Note: A general principal that one can take on here, as an arguable truth about
```
Comment (thomastuts) [inline README.md]: principal -> principle?

File: README.md [hunk 3]
```diff
@@ -1,30 +1,177 @@
...
+    fetch('/api/fruit=?eaten=true').then((data) => {
+      this.setState({ fruitList: data });
+    });
+  }
+
+  render() {
+    return (
+      <ul>
+        {this.state.fruitList.map((fruit) => {
+          return <li key={fruit}>{fruit}</li>;
```
Comment (thomastuts) [inline README.md]: Maybe use the arrow function shorthand here to showcase even more of ES2015's awesomeness? Minor suggestion.

File: README.md [hunk 4]
```diff
@@ -1,30 +1,177 @@
...
+and update its state. Already, we can see that we're starting to repeat ourselves
+unnecessarily. The fetching process for fruit isn't really all that different in 
+each component. So why do we need to implement it twice? Plus, in general, we know
+that we should minimize state as much as possible to avoid bugs, and yet here we
+have two components maintaining their own state. That's a lot of state in such a
+small app!
+
+## How to minimize state with `props`
+
+So what can we do to minimize state? Well, let's think. Really, our two components
```
Comment (thomastuts) [inline README.md]: I think we can ditch the first 'Really'

File: README.md [hunk 5]
```diff
@@ -1,30 +1,177 @@
...
+are both lists; they aren't really all that different. What if we just
+conceive of our two components as one simple "presentational" or "dumb" component
```
Comment (thomastuts) [inline README.md]: 'conceive of' sounds a bit weird, maybe something like 'think of'?

File: README.md [hunk 6]
```diff
@@ -1,30 +1,177 @@
...
+    this.updateFruitList = this.updateFruitList.bind(this);
+    this.toggleListMode = this.toggleListMode.bind(this);
+  }
+
+  componentWillMount() {
+    this.updateFruitList();
+  }
+
+  toggleListMode() {
+    this.set({ showEaten: !this.state.showEaten });
```
Comment (thomastuts) [inline README.md]: `this.set` -> `this.setState`

File: README.md [hunk 7]
```diff
@@ -1,30 +1,177 @@
...
+      this.setState({ fruits: data });
+    });
+  }
+
+  render() {
+    render <FruitList fruitList={this.state.fruits} />;
+  }
+}
+```
+So what have we done here? Well, there's a bunch of code here, but really all
```
Comment (thomastuts) [inline README.md:151]: Maybe leave out 'really' to improve the flow?

File: README.md [hunk 8]
```diff
@@ -1,30 +1,177 @@
...
+we've done is setup the same mechanism to fetch a list of fruit from the backend.
+We necessarily had to add a bit more logic to switch the application's "mode"
+so that the `App` component fetches the appropriate list, but then all we had
+to do is send that list down into our `FruitList` component to be rendered.
+
+Now this might seem as though it's a bit more involved, but think of what we've
+accomplished here! Let's summarize. By using props to send our fruit list data
+downward, we've been able simplify our application's implementation, reducing
```
Comment (thomastuts) [inline README.md]: able **to**

File: README.md [hunk 9]
```diff
@@ -1,30 +1,177 @@
...
+We've also been able to remove all that complex state logic, once maintained in 
+our `FruitsEaten` and `FruitsNotEaten` components, by _pushing
+it up the component hierarchy_. Because our newly simplified `FruitList` component
+just receives the list of fruits it should display from its parent component, its
+parent component `App` is the only one that needs to be "smart". So not only have
+we reduced our code base, we've also centralized  our business logic, which will
+make it much easier to modify and maintain going forward. This is a huge gain --
+all thanks to the power of `props`.
+
+Mad props to props!
```
Comment (thomastuts) [inline README.md:172]: 🎉

**Comment by thomastuts** [comment]
This is great! Some minor notes here and there, but other than that it looks awesome! 👍

File: README.md (hunk 7 above)
Comment (ezmiller) [reply README.md:151]: I think the word is appropriate here. It minimizes the significance of the changes, which is what I wanted to achieve.

**Comment by ezmiller** [comment]
@thomastuts changes made. Let me know if i should merge.

**Comment by thomastuts** [comment]
@ezmiller Awesome changes, LGTM! :shipit:

**Pull request merged**